src/Features/CSharp/Portable/UsePatternCombinators/AnalyzedPattern.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UsePatternCombinators
+{
+    /// <summary>
+    /// Base class to represent a pattern constructed from various checks
+    /// </summary>
+    internal abstract class AnalyzedPattern
+    {
+        private AnalyzedPattern()
+        {
+        }
+
+        /// <summary>
+        /// Base class for patterns that target a specific expression, i.e. non-combinators
+        /// </summary>
+        internal abstract class Test : AnalyzedPattern
+        {
+            public readonly IOperation TargetOperation;
```

Comment:
do we really need this type?  could we not move this IOp up to AnalzedPattern?

Reply:
Pattern combinators don't have a target expression, the're only a container, hence the name.